addons/analytic/models/analytic.py

```diff
@@ -19,27 +19,22 @@ class account_analytic_account(models.Model):
 
     @api.multi
     def _compute_debit_credit_balance(self):
-        analytic_line_obj = self.env['account.analytic.line']
-        domain = [('account_id', 'in', self.mapped('id'))]
+        domain = [('account_id', 'in', self.ids),
+                  ('company_id', '=', self.env.user.company_id.id)]
         if self._context.get('from_date', False):
-            domain.append(('date', '>=', self._context['from_date']))
+            domain += [('date', '>=', self._context['from_date'])]
         if self._context.get('to_date', False):
-            domain.append(('date', '<=', self._context['to_date']))
-
-        account_amounts = analytic_line_obj.search_read(domain, ['account_id', 'amount'])
-        account_ids = set([line['account_id'][0] for line in account_amounts])
-        data_debit = {account_id: 0.0 for account_id in account_ids}
-        data_credit = {account_id: 0.0 for account_id in account_ids}
-        for account_amount in account_amounts:
-            if account_amount['amount'] < 0.0:
-                data_debit[account_amount['account_id'][0]] += account_amount['amount']
-            else:
-                data_credit[account_amount['account_id'][0]] += account_amount['amount']
+            domain += [('date', '<=', self._context['to_date'])]
+        amounts = self.env['account.analytic.line'].read_group(
+            domain, ['account_id', 'amount'], ['account_id'])
+        amounts = dict([(amount['account_id'][0], amount['amount'])
+                       for amount in amounts])
 
         for account in self:
-            account.debit = abs(data_debit.get(account.id, 0.0))
-            account.credit = data_credit.get(account.id, 0.0)
-            account.balance = account.credit - account.debit
+            amount = amounts.get(account.id, 0.0)
+            account.balance = amount
+            account.credit = amount if amount > 0.0 else 0.0
+            account.debit = -amount if amount < 0.0 else 0.0
```

Comment:
Oh wait, I think I missed this. It seems you are replacing the sums of debit and credit amounts with the balance. That is not good. What you typically like to see is the debit and credit of an account which represent the expenses and income, and then the balance which represents the profit and loss. By summing debit and credit amongst each other we lose the first type of information.